### PR TITLE
feat(server): Add a way to listen for one request.

### PR DESCRIPTION
**Pull request by reem**

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -91,6 +92,21 @@ S: NetworkStream + Clone + Send> Server<L> {
...
+    /// Blocks until one connection is accepted and a single request is handled.
+    // Uses the FnOnce trait instead of Handler since the ownership mode is different.
+    pub fn oneshot<T, F: FnOnce(Request, Response) -> T>(self, handler: F) -> HttpResult<T> {
+        let stream = try!(self.listener((self.ip, self.port)).accept());
+
+        let mut rdr = BufferedReader::new(stream.clone());
+        let mut wrt = BufferedWriter::new(stream);
+
+        let mut res = Response::new(&mut wrt);
+        let mut req = try!(Request::new(&mut rdr, addr)); // TODO: Send 4XX
```
Comment (seanmonstar) [inline src/server/mod.rs]: Build fails cause `addr` doesn't exist.

**Comment by seanmonstar** [comment]
Needs a rebase

**Comment by reem** [comment]
Does it? I think something is just wrong with a test but I haven't had time to fix it.

**Comment by seanmonstar** [comment]
Well, perhaps not a rebase, but it uses older code (not passing a `&mut NetworkStream`).

**Comment by seanmonstar** [comment]
Closing as stale.

**Pull request closed without merging**